Rebus.AdoNet/AdoNetTimeoutStorage.cs

```diff
@@ -134,11 +145,17 @@ private static string FormatGetTimeoutsDueQuery(SqlDialect dialect, string table
 				lockPredicate = $"AND " + dialect.FormatTryAdvisoryLock(new[] { "id" });
 			}
 
+			if(limit != null)
```

Comment:
If 'limit' is 0.. this will not work. It would be better to avoid declaring 'limit' as nullable, and assume 0 as 'unlimited'.